scripts/merge_changelogs.sh

```diff
@@ -1,14 +1,42 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-# skip the template file
-echo "**Features**"
-grep -i '^Implements:' .unreleased/* | grep -v '.unreleased/template.rfc822' | cut -d: -f3- | sort | uniq | sed -e 's/^[[:space:]]*//' -e 's/^/* /'
-echo
+#
+# This script build a CHANGELOG.md entry for a new release
+#
 
-echo "**Bugfixes**"
-grep -i '^Fixes:' .unreleased/* | grep -v '.unreleased/template.rfc822' | cut -d: -f3- | sort | uniq | sed -e 's/^[[:space:]]*//' -e 's/^/* /'
-echo
+RELEASE_NOTES_TEMPLATE='.unreleased/RELEASE_NOTES.md.j2'
 
-echo "**Thanks**"
-grep -i '^Thanks:' .unreleased/* | grep -v '.unreleased/template.rfc822' | cut -d: -f3- | sort | sed -e 's/^[[:space:]]*//' -e 's/^/* /'
-echo
+echo_changelog() {
+    echo "${1}"
+    # skip the template and release notes files
+    grep -i "${2}" .unreleased/* | \
+        grep -v '.unreleased/template.*' | \
+        grep -v "${RELEASE_NOTES_TEMPLATE}" | \
+        cut -d: -f3- | sort | uniq | sed -e 's/^[[:space:]]*//' -e 's/^/* /'
+    echo
+}
+
+get_version_config_var() {
+    grep "${1}" version.config | awk '{print $3}' | sed 's/-dev//'
+}
+
+RELEASE_CURRENT=$(get_version_config_var '^version')
+RELEASE_PREVIOUS=$(get_version_config_var '^update_from_version')
+RELEASE_DATE=$(date +"%Y-%m-%d")
+
+#
+# To install jinja template client:
+#   $ pip install jinja-cli
+#
+if [ -f "${RELEASE_NOTES_TEMPLATE}" ];
+then
+    jinja \
+        -D release_current "${RELEASE_CURRENT}" \
+        -D release_previous "${RELEASE_PREVIOUS}" \
+        -D release_date "${RELEASE_DATE}" ${RELEASE_NOTES_TEMPLATE}
+    echo
+fi
+
+echo_changelog '**Features**' '^Implements:'
```

Comment:
Here we can have a separate "Major Feature" field or something similar as a separate section.

Reply:
This is something that we can improve in the future. In the current release process we always mention major features in the release notes header.